src/main/java/DataUtils.java

```diff
@@ -0,0 +1,31 @@
+import java.util.Arrays;
+
+public class DataUtils {
+
+    private static final String SUM = "+";
+    private static final String SUBTRACT = "-";
+    private static final String MULTIPLY = "*";
+    private static final String DIVIDE = "/";
+
+    public static String[] calculateData(String[] data) {
+        String[] calculatedData = Arrays.copyOf(data, data.length);
+        for (int i = 0; i < data.length; i++) {
+            String[] separatedData = data[i].split(" ");
+            calculateAndSaveResult(calculatedData, i, separatedData);
```

Comment:
To nie jest błąd, ale ja bym to raczej rozpisał jako:
```
String equation = getEquationWithResult(separatedData);
calculatedData[i] = equation;
```
Takie przekazywanie tablicy i zmienianie jej stanu w jakiejś innej metodzie raczej nie jest spotykane.